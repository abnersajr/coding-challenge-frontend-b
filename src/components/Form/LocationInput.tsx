import React, { useState } from 'react'

import { FormField, FormFieldProps, TextInput, TextInputProps } from 'grommet'

type Props = {
  formField: FormFieldProps
  textInput: TextInputProps
}

const LocationInput = (props: Props) => {
  const [value, setValue] = useState('')

  const onSelect: TextInputProps['onSelect'] = (event) =>
    setValue(event.suggestion)

  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value: newValue } = event.target
    setValue(newValue)

    if (!newValue.trim()) {
      // setSuggestedFolks([])
    } else {
      // // simulate an async call to the backend
      // setTimeout(() => setSuggestedFolks(folks), 300)
    }
  }

  return (
    <FormField {...props.formField}>
      <TextInput
        value={value}
        onChange={onChange}
        onSelect={onSelect}
        {...props.textInput}
      />
    </FormField>
  )
}

export default LocationInput
