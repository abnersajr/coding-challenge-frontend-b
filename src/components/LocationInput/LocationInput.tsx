import React, { useState, useRef } from 'react'

import { FormField, FormFieldProps, TextInput, TextInputProps } from 'grommet'

type Props = {
  formField: FormFieldProps
  textInput: TextInputProps & {
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  }
}

const LocationInput = (props: Props) => {
  return (
    <FormField {...props.formField}>
      <TextInput
        readOnly={true}
        value={props.textInput.value}
        {...props.textInput}
        onChange={props.textInput.onChange}
        onSelect={props.textInput.onSelect}
      />
    </FormField>
  )
}

export default LocationInput
