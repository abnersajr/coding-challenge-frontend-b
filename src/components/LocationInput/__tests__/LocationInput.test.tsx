import React from 'react'
import LocationInput from '../LocationInput'
import { render } from 'test/components'
import { screen, fireEvent } from '@testing-library/react'

describe('LocationInput', () => {
  const props = {
    textInput: {
      value: 'Test Value',
      onChange: jest.fn(),
      onSelect: jest.fn(),
    },
  }
  it('render properly', () => {
    render(<LocationInput {...props} />)
    expect(screen.getByTestId('LOCATION.INPUT')).toMatchSnapshot()
  })
})
