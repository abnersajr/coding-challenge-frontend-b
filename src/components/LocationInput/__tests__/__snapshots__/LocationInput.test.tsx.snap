// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInput render properly 1`] = `
<input
  autocomplete="off"
  class="StyledTextInput-sc-1x30a0s-0 kgYyVh"
  data-testid="LOCATION.INPUT"
  readonly=""
  value="Test Value"
/>
`;
